File: Sistema.java
import java.util.*;

public class Sistema {
    Scanner sc = new Scanner(System.in);
    private final String contraseña = "Psw/8.";
    Tienda tienda = new Tienda();

    public void ejecutarPrograma(){
        printHeader("BIENVENIDO");
        while(true){
            System.out.println("Por favor ingrese la contraseña: ");
            System.out.print(">> ");
            String intento = sc.nextLine();
            if(intento.equals(contraseña)){
                System.out.println("¡Contraseña correcta!");
                System.out.print("Presione enter para continuar...");
                sc.nextLine();
                ejecutarSistema();
                break;
            }else{
                System.out.println("Contraseña incorrecta");
            }
        }   
    }

    private void ejecutarSistema(){
        tienda.addCliente(new Cliente("Jorge", "CURP" , "25/03/03" , 123456789));
        tienda.addLimp(new Limpieza("Lysol", 1534, 15.32, "25/03/23", 15, "Lysol"));
        int opc = 0;

        while(opc<11){
            printHeader("SISTEMA");
            System.out.println("Seleccione una opción:");
            System.out.println("1. Registrar producto");
            System.out.println("2. Registrar cliente");
            System.out.println("3. Mostrar productos");
            System.out.println("4. Mostrar clientes");
            System.out.println("5. Eliminar producto");
            System.out.println("6. Eliminar cliente");
            System.out.println("7. Añadir a Stock");
            System.out.println("8. Eliminar del Stock");
            System.out.println("9. Realizar una compra");
            System.out.println("10. Ver compras");
            System.out.println("11. Salir");
            System.out.print(">> ");
            opc = sc.nextInt();
            sc.nextLine();

            switch (opc) {
                case 1:
                    tienda.registrarProducto();
                    break;

                case 2:
                    tienda.registrarCliente();
                    break;
            
                case 3:
                    tienda.mostrarProductos();
                    break;
            
                case 4:
                    tienda.mostrarClientes();
                    break;
            
                case 5:
                    tienda.eliminarProducto();
                    break;
            
                case 6:
                    tienda.eliminarCliente();
                    break;
            
                case 7:
                    tienda.añadirStock();
                    break;
            
                case 8:
                    tienda.quitarStock();
                    break;
            
                case 9:
                    tienda.realizarCompra();
                    break;
            
                case 10:
                    tienda.verCompras();
                    break;
            
                default:
                    break;
            }
        }
    }
    
    private void printHeader(String header){
        System.out.println("===================================================");
        System.out.println(header);
        System.out.println("===================================================");
    }
}

File: Sistema.py
from Tienda import Tienda
from Cliente import Cliente
from Limpieza import Limpieza

import getpass

class Sistema:
    def __init__(self):
        self.contraseña = "Psw/8."
        self.tienda = Tienda()

    def ejecutarPrograma(self):
        self.printHeader("BIENVENIDO")
        while True:
            intento = input("Por favor ingrese la contraseña: ")
            if intento == self.contraseña:
                print("¡Contraseña correcta!")
                input("Presione enter para continuar...")
                self.ejecutarSistema()
                break
            else:
                print("Contraseña incorrecta")

    def ejecutarSistema(self):
        self.tienda.addCliente(Cliente("Jorge", "CURP", "25/03/03", 123456789))
        self.tienda.addProducto(Limpieza("Lysol", 1534, 15.32, "25/03/23", 15, "Lysol"))
        opc = 0

        while opc < 11:
            self.printHeader("SISTEMA")
            print("Seleccione una opción:")
            print("1. Registrar producto")
            print("2. Registrar cliente")
            print("3. Mostrar productos")
            print("4. Mostrar clientes")
            print("5. Eliminar producto")
            print("6. Eliminar cliente")
            print("7. Añadir a Stock")
            print("8. Eliminar del Stock")
            print("9. Realizar una compra")
            print("10. Ver compras")
            print("11. Salir")
            opc = int(input(">> "))

            if opc == 1:
                self.tienda.registrarProducto()
            elif opc == 2:
                self.tienda.registrarCliente()
            elif opc == 3:
                self.tienda.mostrarProductos()
            elif opc == 4:
                self.tienda.mostrarClientes()
            elif opc == 5:
                self.tienda.eliminarProducto()
            elif opc == 6:
                self.tienda.eliminarCliente()
            elif opc == 7:
                self.tienda.añadirStock()
            elif opc == 8:
                self.tienda.quitarStock()
            elif opc == 9:
                self.tienda.realizarCompra()
            elif opc == 10:
                self.tienda.verCompras()

    def printHeader(self, header):
        print("===================================================")
        print(header)
        print("===================================================")